accounts/abi/unpack.go

```diff
@@ -43,43 +44,72 @@ var (
 )
 
 // ReadInteger reads the integer based on its kind and returns the appropriate value.
-func ReadInteger(typ Type, b []byte) interface{} {
+func ReadInteger(typ Type, b []byte) (interface{}, error) {
```

Comment:
👀

Reply:
need to double check that this is backwards compatible where we use the ABI package

Reply:
looking into this